cap-primitives/src/fs/metadata.rs

```diff
@@ -198,129 +198,191 @@ impl Metadata {
     }
 }
 
+/// Unix-specific extensions for [`MetadataExt`].
+///
+/// This corresponds to [`std::os::unix::fs::MetadataExt`].
+#[cfg(any(unix, target_os = "vxworks"))]
+pub trait MetadataExt {
+    /// Returns the ID of the device containing the file.
+    fn dev(&self) -> u64;
+    /// Returns the inode number.
+    fn ino(&self) -> u64;
+    /// Returns the rights applied to this file.
+    fn mode(&self) -> u32;
+    /// Returns the number of hard links pointing to this file.
+    fn nlink(&self) -> u64;
+    /// Returns the user ID of the owner of this file.
+    fn uid(&self) -> u32;
+    /// Returns the group ID of the owner of this file.
+    fn gid(&self) -> u32;
+    /// Returns the device ID of this file (if it is a special one).
+    fn rdev(&self) -> u64;
+    /// Returns the total size of this file in bytes.
+    fn size(&self) -> u64;
+    /// Returns the last access time of the file, in seconds since Unix Epoch.
+    fn atime(&self) -> i64;
+    /// Returns the last access time of the file, in nanoseconds since [`atime`].
+    fn atime_nsec(&self) -> i64;
+    /// Returns the last modification time of the file, in seconds since Unix Epoch.
+    fn mtime(&self) -> i64;
+    /// Returns the last modification time of the file, in nanoseconds since [`mtime`].
+    fn mtime_nsec(&self) -> i64;
+    /// Returns the last status change time of the file, in seconds since Unix Epoch.
+    fn ctime(&self) -> i64;
+    /// Returns the last status change time of the file, in nanoseconds since [`ctime`].
+    fn ctime_nsec(&self) -> i64;
+    /// Returns the block size for filesystem I/O.
+    fn blksize(&self) -> u64;
+    /// Returns the number of blocks allocated to the file, in 512-byte units.
+    fn blocks(&self) -> u64;
+    #[cfg(target_os = "vxworks")]
```

Comment:
It seems easy for us to somehow miss future extensions to these traits in std, especially slightly more obscure OSes.  This seems to be the main downside of this.  But eh, in the end if there's some obscure new API and we don't notice, it can just be added.

Reply:
Yeah. All of cap-std already faces this challenge; it's just the nature of what cap-std being what it is.